storage/indexer/src/db_indexer.rs

```diff
@@ -395,11 +403,15 @@ impl DBIndexer {
                             .expect("Failed to put state keys to a batch");
                     }
                 });
+                batch.put::<InternalIndexerMetadataSchema>(
+                    &MetadataKey::StateVersion,
+                    &MetadataValue::Version(version),
+                )?;
             }
             version += 1;
             Ok::<(), AptosDbError>(())
         })?;
-        assert_eq!(num_transactions, version - start_version.unwrap_or(0));
+        assert_eq!(num_transactions, version - start_version);
         batch.put::<InternalIndexerMetadataSchema>(
```

Comment:
need to put in per enabled data type progress here

Reply:
Done